### PR TITLE
[IMP] web: Advanced Search (+ other small refs)

### PR DESCRIPTION
**Pull request by Polymorphe57**
The old submenu "Add Custom Filter" of the menu "Filters" is now replaced
by a menu item "Advanced Search". Click on that item will open an
"Advanced Search" dialog in which the current search domain is displayed
in a domain selector. It is then possible to easily add/remove/edit some
parts of that domain before make a new search with the edited domain.

Task ID: 3269923

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]

File: addons/web/static/src/search/filter_menu/advanced_search_dialog.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/** @odoo-module **/
+
+import { _t } from "@web/core/l10n/translation";
+import { Component } from "@odoo/owl";
+import { Domain } from "@web/core/domain";
+import { DomainSelectorDialog } from "@web/core/domain_selector_dialog/domain_selector_dialog";
+import { SearchModel } from "../search_model";
+import { useService } from "@web/core/utils/hooks";
+
+class ModDomainSelectorDialog extends DomainSelectorDialog {
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/web/static/src/search/filter_menu/advanced_search_dialog.js:10]: For modified?!

File: addons/web/static/src/core/domain_selector/domain_selector.xml [hunk 2]
```diff
@@ -2,57 +2,55 @@
...
-                    <button class="btn btn-sm btn-primary o_domain_add_first_node_button ms-1" t-on-click="() => this.insertRootLeaf(node)">
-                        <i class="fa fa-plus"/> Add filter
-                    </button>
+                    <button class="btn btn-sm btn-primary o_domain_add_first_node_button ms-1" t-on-click="() => this.insertRootLeaf(node)">Reset domain</button>
+                </t>
+                <t t-if="props.isDebugMode">
+                    <label class="o_domain_debug_container">
+                        <span class="small"># Code editor</span>
+                        <textarea type="text" class="o_domain_debug_input" t-att-readonly="props.readonly" spellcheck="false" t-att-value="props.debugValue or props.value" t-on-change="(ev) => this.onDebugValueChange(ev.target.value)" />
+                    </label>
```
Comment (mcm-odoo) [inline addons/web/static/src/core/domain_selector/domain_selector.xml:19]: extract in a new template?

File: addons/web/static/src/search/filter_menu/advanced_search_dialog.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (aab-odoo) [reply addons/web/static/src/search/filter_menu/advanced_search_dialog.js:10]: Why modified? I don't really like the name

**Comment by ged-odoo** [comment]
robodoo r+

**Comment by pedrobaeza** [comment]
Thanks for the improvement.

I see that finally the = option for m2x fields is not showing a dropdown for selecting the value(s) in a similar manner to the many2one widget. Is that something planned for a future PR or is it going to be left as it is?

cc @bouvyd, as I talked with him about this.

**Comment by Polymorphe57** [comment]
> 

Hello. This is not yet planned (at my best knowledge). But I also think it would be nice to have that. There are other PRs that should be soon merged and that will bring many other improvements to DomainSelector, ModelFieldSelector,... Maybe I will be able to make that available at some point.

**Comment by pedrobaeza** [comment]
Great, it would be a very good complement.

Congratulations anyway on the new feature!

**Comment by bouvyd** [comment]
Hello,

Yes it is planned, but as-of yet not spec'ed. Waiting for the advanced search to be fully done before moving forward with small refinements (including allowing to select values for m2o or m2m, for example). No promises until anything is merged though, as you know :)

**Pull request closed without merging**